node/pkg/node/node.go

```diff
@@ -240,6 +268,32 @@ func GuardianOptionWatchers(watcherConfigs []watchers.WatcherConfig, ibcWatcherC
 				}(chainMsgC[chainId], chainId)
 			}
 
+			// Per-chain query response channel
+			chainQueryResponseC := make(map[vaa.ChainID]chan *query.PerChainQueryResponseInternal)
+			// aggregate per-chain msgC into msgC.
+			// SECURITY defense-in-depth: This way we enforce that a watcher must set the msg.EmitterChain to its chainId, which makes the code easier to audit
+			for _, chainId := range vaa.GetAllNetworkIDs() {
+				chainQueryResponseC[chainId] = make(chan *query.PerChainQueryResponseInternal)
+				go func(c <-chan *query.PerChainQueryResponseInternal, chainId vaa.ChainID) {
```

Comment:
Since this collates all the chain channels.  Should we be concerned if only one goes away?